### ISSUE TITLE
[onNewMessage] on new RequestHandler function (deep-chat-dev, react-dev 9.0.5)

### ISSUE THREAD

**Issue by GIOVANNI-75** [OvidijusParsiunas/deep-chat#16]
hello, I saw that you had finished updating the main branch, it's a good job, I tested the new functionality and I congratulate you,
I encountered a small problem when finishing a project with the new requestHandler, everything works well except that it does not use the same signal as the base URL, therefore the interception functions like onNewMessage or even the body prop returned by the requestHandler itself does not intercept incoming messages, before  was intercepted with the signal directly and as now it is a separate logic I think that incoming messages would be taken from the front directly but I think that will slow down the process a bit , I also tried to update the "initialMessages" to insert the incoming message each time but once uploaded we cannot change it, tell me what you think about it

**Comment by GIOVANNI-75** [comment]
I wrote a little tip to get around the problem but tell me if there is anything specific to do about it

**Comment by OvidijusParsiunas** [comment]
Hi, thankyou for raising this request.
I apologise but I don't think I fully understand your first comment. Do you mean that the `onNewMessage` event is giving you the incorrect message information?
I have tested the interceptors and they appear to work ok. The expected order that the interceptors are meant to be executed in is `requestInterceptor` -> `handler` -> `responseInterceptor`. (`responseInterceptor` is not available for streams)

Could you elaborate on the specific details in your workflow that are causing the problem and provide a coding example to help me understand better. Thanks!

**Comment by GIOVANNI-75** [comment]
messages rendered by the logic written within the handler function: are not taken into account by onNewMessage event,
and also in the body of (handler: ("body", signal)) the messages belonging to [role: "ai"] are empty.
messages returned by the Handler function (stream={true}) are not taken into account by Events

**Comment by OvidijusParsiunas** [comment]
I have created a stackblitz codebase where everything appears to work fine, could you maybe edit it or comment with the code that would cause the problem? Thanks!

https://stackblitz.com/edit/react-zdiigp?file=src%2FApp.js

**Comment by OvidijusParsiunas** [comment]
(It is nearly 4am where I live so I will go to bed and try to help you as soon as I get up, thanks!)

**Comment by GIOVANNI-75** [comment]
I admit that on stackblitz it works very well but with me it does not work I tested locally, server, and in build in strict and normal mode, I count two bugs, the first and the body sent back by Handler Request does not include the incoming role:ai message and the second bug is at the front-end level, the message rendered is doubled for the old message
<img width="372" alt="requestHandler function example" src="https://github.com/OvidijusParsiunas/deep-chat/assets/58960396/3235e120-8514-4739-b710-d697df260c41">
<img width="854" alt="Browser Log for Body messages render" src="https://github.com/OvidijusParsiunas/deep-chat/assets/58960396/c8298077-2d1d-4cbb-904d-08e7aec86ad1">

**Comment by OvidijusParsiunas** [comment]
Hi @GIOVANNI-75, thankyou for sharing your examples.
I think I see the problem, I will try to replicate it locally.
It is worth mentioning that the `handler` functionality is supported in package version `1.3.2` and you are using `1.3.0`. Can you try to update to the latest version to see if it helps fix the issue?

**Comment by GIOVANNI-75** [comment]
I was wrong in my last message, it's version 1.3.2 that I'm using
![IMG_1752](https://github.com/OvidijusParsiunas/deep-chat/assets/58960396/bbcf7fc1-529e-4590-a143-91d5b6d60c46)

**Comment by OvidijusParsiunas** [comment]
I have attempted to replicate the problem but everything appears to be working fine for me. I did notice a small potential problem where the `messages` array that we pass to the handler is the actual reference of what the component is using - so manipulating that would can cause issues.
I have fixed this bug in `deep-chat-dev` and `deep-chat-react-dev` packages version `9.0.7`, could you try to see if that perhaps helps.

If the issue is still there, then I would need to see the full configuration you are using for deep-chat in your app. Additionally, I am not sure what library you are using for `this.sleep` syntax, but if you could perhaps try to temporarily remove it and see if it helps.
E.g. does this work?

```
handler: async (body, signals) => {
  console.log(body.messages);
  signals.onOpen();
  signals.onResponse({text: 'text'});
  signals.onClose();
}
```

Let me know what the outcome is, thanks!

**Comment by GIOVANNI-75** [comment]
hey , after some tests I realized that the problem comes from the fact that I use DeepChat in an App class extends React and not an App function

**Comment by GIOVANNI-75** [comment]
do you think that when we are in a Class the rendering of the props is not updated?

**Comment by GIOVANNI-75** [comment]
hey, the message duplication problem comes from the Avatar prop "avatars={avatars}"

**Comment by OvidijusParsiunas** [comment]
Oh wow, that is a really good catch! Thanks for letting me know. It appears this is also a problem for the `names` property too. I have deployed a hot-fix for this in `deep-chat-dev` and `deep-chat-react-dev` version `9.0.9` and will get it out to the core package tomorrow. Could you check if perhaps this also fixes your other issue?

If not, could you share other properties you are using in your component. Thanks!

**Comment by GIOVANNI-75** [comment]
perfect. it works very well 👍🏼

**Comment by OvidijusParsiunas** [comment]
Good to hear, does it fix both of the problems you were experiencing?

**Comment by OvidijusParsiunas** [comment]
The bug fix is now available in the core version `1.3.4`. Thankyou once again for letting me know about it. I will close this issue, but if you are experiencing any other problems let me know. Thanks!!
